src/_ert_job_runner/client.py

```diff
@@ -74,18 +86,31 @@ async def _send(self, msg: AnyStr) -> None:
                 await self.websocket.send(msg)
```

Comment:
It's hard to say which exceptions belong to `get_websocket` and which to `send()`. Is it an idea to split the `try:` block?

Reply:
@lars-petter-hauge provided a version, which solves the per-partes try: except: but we both agreed it's not looking as good :) 
```python
async def _handle_exception(self, msg, exc, retry):
    if retry == self._max_retries:
        logger.debug(msg)
        raise ClientConnectionError(msg) from exc
    await asyncio.sleep(0.2 + self._timeout_multiplier * retry)

async def _send(self, msg: AnyStr) -> None:
    for retry in range(self._max_retries + 1):
        if self.websocket is None:
            try:
                self.websocket = await self.get_websocket()
            except (
                InvalidHandshake,
                InvalidURI,
                asyncio.TimeoutError,
                ConnectionRefusedError,
                OSError,
            ) as exception:
                _error_msg = (
                    f"Not able to establish the "
                    f"websocket connection {self.url}! Max retries reached!"
                )
                await self._handle_exception(_error_msg, exception, retry)
                continue
        try:
            await self.websocket.send(msg)
        except ConnectionClosedOK:
            logger.debug("Connection was closed from the server with ok message!")
            raise
        except ConnectionClosedError as exception:
            _error_msg = (
                f"Not been able to send the event"
                f" to {self.url}! Max retries reached!"
            )
            await self._handle_exception(_error_msg, exception, retry)
```

Reply:
I think this can be split without increasing indendation. But splitting it makes `_send` hard to read. Can we push the exception handling for `get_websocket()` into `get_websocket()` itself? (it could even has its own retry loop)

Reply:
We considered it but this would change the behavior of `get_websockets()` since we won't to rise `ClientConnectionError` instead.